src/host/srvinit.cpp

```diff
@@ -653,22 +653,22 @@ DWORD ConsoleIoThread()
         if (ReplyMsg != nullptr)
         {
             LOG_IF_FAILED(ReplyMsg->ReleaseMessageBuffers());
-        }

```

Comment:
I'm not so sure about this one. I think there are API calls that don't return a `ReplyMsg`. If any of those are called, then we would be stuck.

Reply:
In that case, with the old code, `ReplyMsg->Complete` would be a null dereference.

Reply:
Should be ok now that I see ReadIo() can take nullptr as 1st parameter.

Reply:
You're right about the null dereference at `&ReplyMsg->Complete`. I looked back at the original code this was adapted from, and what should happen is:

1. ReadIo's signature should be taking the PCONSOLE_API_MSG ReplyMsg as parameter one and only attempting to dig out the completion part if it is not null at that level.
2. We need to keep the } on line 656 because there's technically a situation where the IO could fail for not pipe-disconnected and try to go around the loop again to reattempt the call with there being no reply message at all.